### ISSUE TITLE
smbus python3 - no option in menu, but package can be compiled

### ISSUE THREAD

**Issue by nekitip** [openwrt/packages#2748]
There is a smbus python package in **i2c-tools.**
The same file (from openwrt) can also be compiled to work with python3 in regular debian x64 by simply "python3 setup.py build" on that same file. However it's near to imposible to do it this way in openwrt toolchain (crosscompile) by hand.
Openwrt makemenu already has option to build package and it builds fine and works, but it builds in python 2.7. So, the only thing needed to change are options for **makemenu**, so python 3.x can utilise this package.
Unfortunately, I have failed to make those changes myself... :(
UPDATE:
after a few days of experimenting, i did manage to compile 3.5.1 i2c driver, however, I've done it dirty by manualy modifying makefile in a way of changing python references to python3 (essentially every pysomething to py3something. So, now I have it compiled, although the build process fails because, IMO wrong setting of instal dir location during this dirty work. I can post this file, but since it's a dirty work, I think it would be better if someone who knows what to do do it.
-Therefore, to build i2c for python 3 the dirty way, first you build your normal image with options you need in make menuconfig, then install that image, and then change i2ctools makefile, build image again, process fails, but you just go to i2ctools build_dir and find there xx35.so file and copy it to your target machine python3.x. Try it by typing "python3" to start interpreter, and then "import smbus".
This can be avoided if there would be a regular option for 3.x python smbus in make menuconfig.
-Do have in mind though, that smbus c source itself is a mystery to me, since eventually I ended up using some patched file from net (you can google that one easily). That may imply that this fix will not be accepted, but like I said - the bigest problem for me was crosscompiling, since that patched source file is somethng you easy find. And it may only be my problem, it may happen that even the original works good in 3.x, but....
Good luck!

**Comment by commodo** [comment]
so, i2c-tools is packaged to work with `python`
to work with python3,it should be packaged to build with python3 bindings; 

it does not look too hard to package it;
but we'd need to get the maintainer to look into it, or you can also update the i2c-tools package and submit a PR
@dangowrt  ?

**Comment by nekitip** [comment]
Like I said - I'm not that experienced. Yes, I can post here the makefile that does compile i2c smbus package into python3 module, but after build, something fails (in my opinion, maybe install dir?), so it has to manually be copied to the right location on target device.
As for smbus c file itself, I'm not sure if mine is not working because I've done somethign wrong in makefile or it is not working at all, because **when you open it** (it's not that big file, try and look at the end) - you will see that someone made a lot of IF python3 cases, wich led me to belive that this .c file can indeed be compiled to python3 without messing with it, but I failed. So, I found on various forums different .c file, sort of a "patched file", that I managed to compile with my changed makefile.
So, if you lost me along the way - here is what I suggest:
-let someone who knows what to do, to first try to create standard openwrt option in makefile envirement for python3 i2c smbus driver, and see if it smbus compiles.
If not, then I will give both my makefile (and you fix it all the way), and this patched .c file I found on various forums.
